extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -413,6 +454,23 @@ public void handle(RoutingContext routingContext) {
         // Headers sent on any request, regardless of the response
         HttpServerCommonHandlers.applyHeaders(httpConfiguration.header, httpRouteRouter);
 
+        if (sessionStore != null) {
+            SessionsConfig sessions = httpConfiguration.sessions;
+            // TODO probably need to normalize and relativize the path here
+            String path = sessions.path;
+            SessionHandler sessionHandler = SessionHandler.create(sessionStore.getValue())
+                    .setSessionTimeout(sessions.timeout.toMillis())
+                    .setMinLength(sessions.idLength)
+                    .setSessionCookiePath(path)
+                    .setSessionCookieName(sessions.cookieName)
+                    .setCookieHttpOnlyFlag(sessions.cookieHttpOnly)
+                    .setCookieSecureFlag(sessions.cookieSecure.isEnabled(httpConfiguration.insecureRequests))
+                    .setCookieSameSite(sessions.cookieSameSite.orElse(null))
+                    .setCookieMaxAge(sessions.cookieMaxAge.map(Duration::toMillis).orElse(-1L));
+            // TODO verify if this is the correct router to which the session handler should be installed
+            httpRouteRouter.route(path).handler(sessionHandler);
```

Comment:
I would say yes. That's the main "public" router. The question, however, is the order of the route.

Reply:
Yes, good point. I need to explore more around this, that's why I added these TODOs.